### DISCUSSION TITLE
implementing a Patch function for notch filtering

### DISCUSSION THREAD

**Discussion by ahmadtourei** [DASDAE/dascore#443] [Ideas]
It would be great to design a notch filter that can remove a narrow frequency band (such as power line noise at 60 Hz). As the processing will be carried out in the frequency domain and also the filter can be a pre-processing step for some data processing pipelines, we'd want to follow the behavior of `Patch.whiten` so the function can be applied on a DFT patch as well to avoid multiple DFT/IDFT. [scipy.signal.iirnotch](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.iirnotch.html#scipy.signal.iirnotch) can be used under the hood.

**Comment by d-chambers** [comment]
Hey @ahmadtourei,

Really good idea!

 ObsPy has a [bandstop](https://docs.obspy.org/packages/autogen/obspy.signal.filter.bandstop.html) implementation that uses a Butterworth-type filter which `Patch.pass_filter` also uses. In fact, I started implementing a [`Patch.stop_filter`](https://github.com/DASDAE/dascore/blob/0022b8fee9a1dfc3f06d846a7643f26a3e6119fe/dascore/proc/filter.py#L176-L194) a few years ago but never actually needed it in my research so I didn't finish it. Ideally, it should have exactly the same arguments as `Patch.pass_filter` but the frequency range specifies the stop, rather than the pass, band.

We could also do a `Patch.notch_filter` or `Patch.irrnotch` if you want to, as it is probably more stable (I guess) for removing very narrow frequencies. Both should be straight-forward to implement using examples already in the proc/filter.py module.

**Comment by ahmadtourei** [comment]
Sounds good! I'll work on it next week and get back to you @d-chambers.

**Comment by ahmadtourei** [comment]
closed by #448
